server/src/main/java/org/elasticsearch/search/fetch/subphase/ExplainPhase.java

```diff
@@ -33,8 +33,8 @@
 public final class ExplainPhase implements FetchSubPhase {
 
     @Override
-    public FetchSubPhaseProcessor getProcessor(SearchContext context) {
-        if (context.explain() == false || context.hasOnlySuggest()) {
```

Comment:
The `hasOnlySuggest` check is now done at the top level

Reply:
slight update; instead of checking for `hasOnlySuggest()`, we check that `query() != null` which is more general.